Disaster_Recovery/recovery_script.ps1

```diff
@@ -0,0 +1,265 @@
+
+    $azLoginMsg = "Please login to Azure CLI.`n" +
+    "`t* If you're running the script for the first time, select yes.`n" +
+    "`t* If you've recently logged in to az while running the script, you can select no.`n" +
+    "Confirm login to azure cli?"
+    if (confirmationPrompt -msg $azLoginMsg) {
+        az login --use-device-code -o table
+    }
+
+    az account set -s $applianceSubscriptionId
+    if ($LASTEXITCODE) {
+        $Error[0] | Out-String >> $logFile
+        throw "The default subscription for the az cli context could not be set."
+    }
+
+    # Add necessary extensions
+    az extension add --upgrade --name arcappliance
+    az extension add --upgrade --name k8s-extension
+    az extension add --upgrade --name customlocation
+    az extension add --upgrade --name connectedvmware
+
+    logH1 "Step 1/5: All extensions successfully installed"
+
+    createRG "$applianceSubscriptionId" "$applianceResourceGroupName"
+
+    logH1 "Step 2/5: Creating the Arc resource bridge"
+    logH2 "Provide vCenter details to deploy Arc resource bridge VM. The credentials will be used by Arc resource bridge to update and scale itself."
+
+    # TODO: Fetch necessary information for all related ARM resources
+    # Get appliance name, cluster extension name, custom location names, vCenter name
+
+    # Delete existing resource bridge VM and ARM resource
+    logH2 "Cleaning up existing Arc Resource Bridge"
+    # Ask if they're sure about deleting
+    $deleteWarningMsg = "This will delete the existing Arc resource bridge and deploy a new one with the existing VM template.`n" +
+    "`t*Confirm deletion and recreation of Arc resource bridge?"
+    if (!(confirmationPrompt -msg $deleteWarningMsg)) {
+        $Error[0] | Out-String >> $logFile
+        throw "Arc resource bridge recovery process canceled."
+    }
+    # TODO: Catch exit code
+    az arcappliance delete vmware --debug --config-file $config
+
+    # Redeploying new resource bridge VM
+    logH2 "Deploying new Arc Resource Bridge"
+    az arcappliance deploy vmware --debug --config-file $config
+
+    # Recreating resource bridge ARM resource and connecting new resource bridge VM and Azure 
+    logH2 "Connecting new Arc Resource Bridge to Azure"
+    az arcappliance create vmware --debug --config-file $config --kubeconfig .\kubeconfig
+
+    $applianceId = (az arcappliance show --subscription $applianceSubscriptionId --resource-group $applianceResourceGroupName --name $applianceName --query id -o tsv 2>> $logFile)
+    if (!$applianceId) {
+        throw "Appliance recreation has failed."
+    }
+    # TODO: Get every 30 seconds, timeout after 5 minutes
+    $applianceStatus = (az resource show --debug --ids "$applianceId" --query 'properties.status' -o tsv 2>> $logFile)
+    if ($applianceStatus -ne "Running") {
+        throw "Appliance is not in running state. Current state: $applianceStatus."
+    }
+
+    logH1 "Step 2/5: Arc resource bridge is back up and running"
```

Comment:
step 3 no ?

Reply:
It marks the completion of step 2, added a 'completed' tag at the end of every step to indicate this